src/bit_encoding/decode.rs

```diff
@@ -781,19 +803,22 @@ mod tests {
     fn root_unit_to_unit() {
         // main = jet_eq_32 :: 2^64 -> 2 # 7387d279
         let justjet = vec![0x6d, 0xb8, 0x80];
-        // Should be able to decode this as a CommitNode...
+        // Should be able to decode this as an expression...
+        let mut iter = BitIter::from(&justjet[..]);
+        decode_expression::<_, Core>(&mut iter).unwrap();
```

Comment:
750abaaa8038045683030c40c0f84da2259ae605: We could even make this `ConstructNode::decode` as a contrast to the other two node types.

Reply:
I'd prefer to pretend that `ConstructNode`s have no canonical encoding, so you have to use this obscure non-exported function to "decode" them.

Though I'm open to arguments otherwise.

Reply:
Well, I guess, eventually we'd like a CLI tool that can decode expressions even if they're not unit->unit.

But we'd have to think a bit about what sharing rules we want to enforce. "No rules" would be simplest and probably makes the most sense. But I think, forbidding sharing of witnesses and disconnects maybe also makes sense. Or forbidding sharing entirely.

Should `ConstructNode`s have a human-readable representation too?

These are all good questions which we should resolve, but I don't think they're high-priority right now and I think there'd be nontrivial work no matter what.

Reply:
This is a big topic, but we definitely need an encoding of raw expressions. People want to share their math gadgets with each other. `ConstructNode::{de,en}encode()` is public so encourage users to use it.

For now, let's merge this PR as is.